vital/types/object_track_set.h

```diff
@@ -118,6 +126,9 @@ class VITAL_EXPORT object_track_set : public track_set
 /// Shared pointer for object_track_set type
 typedef std::shared_ptr< object_track_set > object_track_set_sptr;
 
+/// Helper to iterate over the states of a track as object track states
+inline auto as_object_track = range::transform( object_track_state::downcast );
```

Comment:
This line of code is failing to build on my CentOS 7 machine, gcc 4.8.5-16

Reply:
I'm not sure if I have a GCC 4.8 machine handy. Does this work?

```
inline decltype( range::transform( object_track_state::downcast ) )
as_object_track = range::transform( object_track_state::downcast );
```

(Is it the `inline` or the `auto` that GCC doesn't like?)

Fortunately I believe this can be worked around, at some aesthetic cost.

Reply:
Here's the error I get on GCC 4.9

```
In file included from /var/jenkins_home/jobs/KwiverLinuxPR/workspace/sprokit/processes/kwiver_type_traits.h:48:0,
                 from /var/jenkins_home/jobs/KwiverLinuxPR/workspace/sprokit/processes/core/print_config_process.cxx:33:
/var/jenkins_home/jobs/KwiverLinuxPR/workspace/vital/types/object_track_set.h:130:13: error: ‘kwiver::vital::as_object_track’ declared as an ‘inline’ variable
 inline auto as_object_track = range::transform( object_track_state::downcast );
```

Reply:
Yeah, apparently `-std=c++11` is insufficient for GCC to complain about this C++17 feature :stuck_out_tongue_closed_eyes:. 7b6f7b6328fe reworks this to use `constexpr` instead.